### ISSUE TITLE
Negative block mask egde case

### ISSUE THREAD

**Issue by miguelvr** [miguelvr/dropblock#8]
**Description:**

When the Bernoulli distribution samples two ones within their own block size range, the block mask gets a negative value.

**Example:**

```python
    db = DropBlock2D(block_size=2, drop_prob=0.1)
    mask = torch.tensor([[[1., 0., 0., 0., 0.],
                          [0., 1., 0., 0., 0.],
                          [0., 0., 0., 0., 0.],
                          [0., 0., 0., 0., 0.],
                          [0., 0., 0., 0., 0.]]])

    block_mask = db._compute_block_mask(mask)
```
`block_mask` value:

```
tensor([[[ 1.,  1.,  1.,  1.,  1.,  1.],
         [ 1.,  0.,  0.,  1.,  1.,  1.],
         [ 1.,  0., -1.,  0.,  1.,  1.],
         [ 1.,  1.,  0.,  0.,  1.,  1.],
         [ 1.,  1.,  1.,  1.,  1.,  1.],
         [ 1.,  1.,  1.,  1.,  1.,  1.]]])
```
